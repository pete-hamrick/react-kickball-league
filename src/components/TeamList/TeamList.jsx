import React from 'react'

export default function TeamList() {
  return (
    <section>
      <h1>Team List View</h1>
    </section>
  )
}
