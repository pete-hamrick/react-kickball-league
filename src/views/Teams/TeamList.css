ul {
  list-style-type: none;
}

.link-back {
  display: flex;
  text-decoration: underline;
  margin-left: 20px;
}

.team-list {
  color: blue;
  font-weight: bold;
}

a {
  color: blue;
}

a:visited {
  color: inherit;
}
