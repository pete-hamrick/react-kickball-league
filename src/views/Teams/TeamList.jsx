import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { getTeams } from '../../services/teams'
import './TeamList.css'

export default function TeamList() {
  const [teams, setTeams] = useState([])
  const [loading, setLoading] = useState(true)

  useEffect(() => {
    async function getTheTeams() {
      const data = await getTeams()
      setTeams(data)
      setLoading(false)
    }

    getTheTeams()
  }, [])

  if (loading) return <h1>Loading teams...</h1>

  return (
    <section className="team-container">
      <p className="link-back">
        <Link to="/">Back to Homepage</Link>
      </p>
      <ul className="team-list" aria-label="Team List">
        {teams.map((team) => (
          <li key={team.id}>
            <Link to={`/teams/${team.id}`}>{team.name}</Link>
          </li>
        ))}
      </ul>
    </section>
  )
}
