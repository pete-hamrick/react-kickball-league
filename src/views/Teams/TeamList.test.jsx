import { render } from '@testing-library/react'
import TeamList from './TeamList'

// this doesn't test much without mocking? just snapshots the loading screen

it('should render a list of the teams', () => {
  const { container } = render(<TeamList />)
  expect(container).toMatchSnapshot()
})
