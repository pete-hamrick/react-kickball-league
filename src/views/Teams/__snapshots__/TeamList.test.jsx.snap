// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of the teams 1`] = `
<div>
  <section>
    <p>
      <a
        href="/"
      >
        Back to Homepage
      </a>
    </p>
    <ul
      aria-label="Team List"
      class="team-list"
    >
      <li>
        <a
          href="/teams/10"
        >
          Acme USA
        </a>
      </li>
      <li>
        <a
          href="/teams/1"
        >
          Bridge City Sneakers
        </a>
      </li>
      <li>
        <a
          href="/teams/3"
        >
          Lakeville Thunderfeet
        </a>
      </li>
      <li>
        <a
          href="/teams/4"
        >
          Mt. Hoodies
        </a>
      </li>
      <li>
        <a
          href="/teams/2"
        >
          Stumptown Lumberjacks
        </a>
      </li>
      <li>
        <a
          href="/teams/5"
        >
          Vancougers
        </a>
      </li>
    </ul>
  </section>
</div>
`;
